clippy_lints/src/redundant_local.rs

```diff
@@ -0,0 +1,110 @@
+use clippy_utils::{diagnostics::span_lint_and_help, is_from_proc_macro, ty::needs_ordered_drop};
+use rustc_hir::{def::Res, BindingAnnotation, ByRef, Expr, ExprKind, HirId, Local, Node, Pat, PatKind, QPath};
+use rustc_lint::{LateContext, LateLintPass, LintContext};
+use rustc_middle::lint::in_external_macro;
+use rustc_session::{declare_lint_pass, declare_tool_lint};
+use rustc_span::symbol::Ident;
+
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for redundant redefinitions of local bindings.
+    ///
+    /// ### Why is this bad?
+    /// Redundant redefinitions of local bindings do not change behavior and are likely to be unintended.
+    ///
+    /// Note that although these bindings do not affect your code's meaning, they _may_ affect `rustc`'s stack allocation.
+    ///
+    /// ### Example
+    /// ```rust
+    /// let a = 0;
+    /// let a = a;
+    ///
+    /// fn foo(b: i32) {
+    ///    let b = b;
+    /// }
+    /// ```
+    /// Use instead:
```

Comment:
nit: I'm not sure whether `use instead` is necessary here. Some lints tend to omit it in some cases like these ("no definition with the same name")

Reply:
I have no idea whether it's consistent with other similar lints, but I think it's unlikely to hurt. Also, I think that, in this particular case, it's helpful because a significant fraction of Rust users probably aren't familiar with the term "binding" or "local", so having an example which makes it clear what exactly it means to have a "redundant redefinition of a binding" makes the lint more accessible.

Reply:
that's fine, was just pointing out that others omit it sometimes